app/models/nearby_facility.rb

```diff
@@ -7,13 +7,44 @@ class << self
     attr_writer :validate_on_load
 
     def query(params)
-      json_from_file = File.read(Rails.root.join('modules', 'va_facilities', 'nearby.json'))
-      mocked_response = JSON.parse(json_from_file)
-      return mocked_response if params[:street_address] && params[:city] && params[:state] && params[:zip]
+      if params[:street_address] && params[:city] && params[:state] && params[:zip]
+        isochrone_response = request_isochrone(params)
+        return get_facilities_in_isochrone(params, isochrone_response)
+      end
 
       NearbyFacility.none
     end
 
+    def request_isochrone(params)
+      params[:drive_time] = '30' unless params[:drive_time]
+      address = "#{params[:street_address]} #{params[:city]} #{params[:state]} #{params[:zip]}"
+      query = {
+        waypoint: address,
+        maxtime: params[:drive_time],
+        timeUnit: 'minute',
+        dateTime: '07:30:00',
+        optimize: 'timeWithTraffic',
+        key: Settings.bing.key
+      }
+      response = Faraday.get "#{Settings.bing.base_api_url}/Isochrones", query
+      response.body
+    end
+
+    def get_facilities_in_isochrone(params, isochrone_response)
+      isochrone = JSON.parse(isochrone_response)['resourceSets'][0]['resources'][0]['polygons'][0]['coordinates'][0]
+      linestring = make_linestring(isochrone)
+      make_polygon = "ST_MakePolygon(ST_GeomFromText('LINESTRING(#{linestring})'))"
+      conditions = "ST_Intersects(#{make_polygon}, ST_MakePoint(long, lat))"
+      facilities_query_base_instance = FacilitiesQuery::Base.new(params)
+      BaseFacility::TYPES.flat_map do |facility_type|
+        facilities_query_base_instance.get_facility_data(conditions, params[:type], facility_type, params[:services])
+      end
+    end
+
+    def make_linestring(polygon)
+      polygon.map{|point| "#{point[1]} #{point[0]}"}.join(',')
```

Comment:
I'm gonna need some comments. I do not know how polygons work

Reply:
I've added a few comments in `get_facilities_in_isochrone`. Let me know if they make it easy to walk through the code.